flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java

```diff
@@ -446,7 +447,7 @@ public void setTolerableCheckpointFailureNumber(int tolerableCheckpointFailureNu
      * @param cleanupMode Externalized checkpoint cleanup behaviour.
      */
     @PublicEvolving
-    public void enableExternalizedCheckpoints(ExternalizedCheckpointCleanup cleanupMode) {
+    public void setExternalizedCheckpointCleanup(ExternalizedCheckpointCleanup cleanupMode) {
```

Comment:
Renaming this is a breaking change. We should probably keep the old method, too, and deprecate it.

Reply:
In fact it broke the python api :)

`
Nov 24 11:28:23 FAILED pyflink/datastream/tests/test_check_point_config.py::CheckpointConfigTests::test_get_set_externalized_checkpoints_cleanup`

Reply:
Woops. Thanks, that's something I am obviously not yet sensitive to. I will keep the old one and let it call a new method.

On the note of `@Public` and `@PublicEvolving`: CheckpointConfig has `@Public`, but some of its methods have `@PublicEvolving`. How does that make sense? On behalf of the former, I would expect that all methods of the class must have stable interfaces.
This might be related to there not being a clear opinion on whether these annotations are "inherited" by the methods.

Reply:
It is allowed for individual methods to have a different stability. Basically, methods inherit the stability of their class unless they are annotated themselves. Otherwise extending public APIs with new methods would be very difficult as you could only do it by immediately having new methods be stable.

Reply:
The Python API issues should be resolved now, there were in fact a few more problems than just this renaming.